apps/openmw/mwworld/store.cpp

```diff
@@ -279,7 +280,7 @@ namespace MWWorld
         if (it != mStatic.end() && Misc::StringUtils::ciEqual(it->second.mId, id)) {
             // delete from the static part of mShared
             typename std::vector<T *>::iterator sharedIter = mShared.begin();
-            typename std::vector<T *>::iterator end = sharedIter + mStatic.size();
+            typename std::vector<T *>::iterator end = sharedIter + std::min(mStatic.size(), mShared.size());
```

Comment:
Looks like a cheap hack to me.

With empty `mShared` the whole `while` loop is useless, so you can wrap the whole `mShared` handling by the `mShared.size() > 0` check. 

But I am not sure if the `mShared` really should be empty here, so the issue may be somewhere else - normally we assume that the `mShared.size() >= mStatic.size()`.

Reply:
Yeah, it really should never happen because mShared should have the same number of elements as mStatic and mDynamic combined. I don't suppose there can be a clear test case?

Reply:
Yeah, agreed on it being a cheap hack. (I really just wanted to check #2529 yesterday.)

Anyway, apparently dialogue records don't get inserted into mShared while loading. Which makes sense because they get copied in after and there'd be no point in inserting them the first time if they're just going to get cleared again right after.